### DISCUSSION TITLE
V3 Report API: SponsoredBrands - sbPurchasedProduct report has no campaign id metric

### DISCUSSION THREAD

**Discussion by papaDhk** [amzn/ads-advanced-tools-docs#97] [Q&A]
Hello,

I generated the new [Sponsored Brands sbPurchasedProduct report using the latest v3 advertising report API](https://advertising.amazon.com/API/docs/en-us/reporting/v3/report-types#sponsored-brands)

However, I noticed that although we have access to the campaignName metric, the crucial campaignId  and adGroupId  metrics are missing. 
It is essential for correctly identifying campaigns and adGroups

Is it normal?

When trying to retrieve the campaignId metric, here is what I have:

` Impossible to request a report - ReasonPhrase: configuration columns includes invalid values: (campaignId). Allowed values:
(date, startDate, endDate, budgetCurrency, campaignBudgetCurrencyCode, campaignName, adGroupName, attributionType, purchasedAsin,
productName, productCategory, sales14d, orders14d, unitsSold14d, newToBrandSales14d, newToBrandOrders14d, newToBrandPurchases14d,
newToBrandUnitsSold14d, newToBrandSalesPercentage14d, newToBrandOrdersPercentage14d, newToBrandPurchasesPercentage14d, 
newToBrandUnitsSoldPercentage14d)`

**Comment by elizafarley** [comment]
Hi @papaDhk, the team is working on adding `campaignId` and `adGroupId` to the `sbPurchasedProduct` report. We will update the [release notes](https://advertising.amazon.com/API/docs/en-us/info/release-notes) when these metrics are available, and I'll also update you here.

**Reply by papaDhk** [reply]
Thank you @elizafarley!

**Comment by elizafarley** [comment]
@papaDhk In case you missed it, I wanted to follow up here and let you know that `campaignId` and `adGroupId` are now available in the sbPurchasedProduct report! 

See the release note for more details: https://advertising.amazon.com/API/docs/en-us/release-notes/index#campaign-and-ad-group-ids-now-available-in-sponsored-brands-purchased-products-api-reports
